packages/mattermost-redux/src/utils/post_utils.ts

```diff
@@ -217,6 +229,11 @@ export function shouldUpdatePost(receivedPost: Post, storedPost?: Post): boolean
             return true;
         }
 
+        if (!storedPost.metadata && receivedPost.metadata) {
```

Comment:
Would there be a situation where metadata would exist in `storedPost` but receivedPost would have updated or different metadata?

Reply:
Yes this is a possibility, but if `receivedPost` has different or updated metadata, then it's `update_at` field should be different as well and `shouldUpdate` will return true. We don't need to consider the case where the post is a permalink because each post is processed individually, so both the original post and permalink will be covered.